open_feature_contrib/providers/flagd/provider.py

```diff
@@ -0,0 +1,186 @@
+"""
+# This is a Python Provider to interact with flagd
+#
+# -- Usage --
+# open_feature_api.set_provider(flagd_provider.FlagdProvider())
+# flag_value =  open_feature_client.get_string_value(
+#                   key="foo",
+#                   default_value="missingflag"
+#               )
+# print(f"Flag Value is: {flag_value}")
+#   OR the more verbose option
+# flag = open_feature_client.get_string_details(key="foo", default_value="missingflag")
+# print(f"Flag is: {flag.value}")
+#   OR
+# print(f"Flag Details: {vars(flag)}"")
+#
+# -- Customisation --
+# Follows flagd defaults: 'http' protocol on 'localhost' on port '8013'
+# But can be overridden:
+# provider = open_feature_api.get_provider()
+# provider.initialise(schema="https",endpoint="example.com",port=1234,timeout=10)
+"""
+
+import typing
+from numbers import Number
+
+import requests
+from open_feature.evaluation_context.evaluation_context import EvaluationContext
+from open_feature.exception.error_code import ErrorCode
+from open_feature.flag_evaluation.flag_evaluation_details import FlagEvaluationDetails
+from open_feature.provider.provider import AbstractProvider
+
+from .defaults import Defaults
+from .evaluation_context_serializer import EvaluationContextSerializer
+from .flag_type import FlagType
+from .web_api_url_factory import WebApiUrlFactory
+
+
+class FlagdProvider(AbstractProvider):
+    """Flagd OpenFeature Provider"""
+
+    def __init__(
+        self,
+        name: str = "flagd",
+        schema: str = Defaults.SCHEMA,
+        host: str = Defaults.HOST,
+        port: int = Defaults.PORT,
+        timeout: int = Defaults.TIMEOUT,
+    ):
+        """
+        Create an instance of the FlagdProvider
+
+        :param name: the name of the provider to be stored in metadata
+        :param schema: the schema for the transport protocol, e.g. 'http', 'https'
+        :param host: the host to make requests to
+        :param port: the port the flagd service is available on
+        :param timeout: the maximum to wait before a request times out
+        """
+        self.provider_name = name
+        self.schema = schema
+        self.host = host
+        self.port = port
+        self.timeout = timeout
+
+        self.url_factory = WebApiUrlFactory(self.schema, self.host, self.port)
+
+    def get_metadata(self):
+        """Returns provider metadata"""
+        return {
+            "name": self.get_name(),
+            "schema": self.schema,
+            "host": self.host,
+            "port": self.port,
+            "timeout": self.timeout,
+        }
```

Comment:
Not sure how we should handle the additional keys, but get_metadata in AbstractProvider is annotated as returning an instance of `open_feature.provider.metadata.Metadata`